src/PIL/TiffImagePlugin.py

```diff
@@ -1142,6 +1142,17 @@ def load_end(self):
         if not self.is_animated:
             self._close_exclusive_fp_after_loading = True
 
+            # reset buffered io handle in case fp
+            # was passed to libtiff, invalidating the buffer
+            self.fp.tell()
```

Comment:
This is copied from https://github.com/python-pillow/Pillow/blob/94ecb279037501e99d1e329634cc1f2cab95cc13/src/PIL/TiffImagePlugin.py#L1075-L1077